### PR TITLE
Keeping randn in TwoQubitWeylDecomposition but seeded

### PR DESCRIPTION
**Pull request by 1ucian0**
This is an alternative to #3652 .

It seems that is important that the attempts are pulled out from a gaussian distribution. I'm trying to understand why by reading arXiv:1811.12926...

### CODE REVIEW

**Comment by levbishop** [comment]
Better to use RandomState() to avoid monkeying with the global RNG state

**Comment by 1ucian0** [comment]
Much better indeed. Thanks!

**Comment by kdk** [comment]
Can you add the failures found by the existing tests as fixed test cases?

File: test/python/quantum_info/test_synthesis.py [hunk 1]
```diff
@@ -542,6 +542,12 @@ def test_cx_equivalence_3cx_random(self):
         U = execute(qc, sim).result().get_unitary()
         self.assertEqual(two_qubit_cnot_decompose.num_basis_gates(U), 3)
 
+    def test_seed_289(self):
+        """This specif case fails when the PR #3585 was applied
```
Comment (ajavadia) [inline test/python/quantum_info/test_synthesis.py]:
```suggestion
        """This specific case failed when PR #3585 was applied
```

File: qiskit/quantum_info/synthesis/two_qubit_decompose.py [hunk 2]
```diff
@@ -161,8 +161,9 @@ def __init__(self, unitary_matrix):
         # M2 is a symmetric complex matrix. We need to decompose it as M2 = P D P^T where
         # P ∈ SO(4), D is diagonal with unit-magnitude elements.
         # D, P = la.eig(M2)  # this can fail for certain kinds of degeneracy
-        for _ in range(100):  # FIXME: this randomized algorithm is horrendous
-            M2real = np.random.randn()*M2.real + np.random.randn()*M2.imag
+        for i in range(100):
```
Comment (1ucian0) [inline qiskit/quantum_info/synthesis/two_qubit_decompose.py]:
```suggestion
        for i in range(100):  # FIXME: this randomized algorithm is horrendous
```

**Comment by 1ucian0** [comment]
> Can you add the failures found by the existing tests as fixed test cases?

Done in e08acbb.

**Comment by 1ucian0** [comment]
I tried this branch with 1 million random unitaries and it worked.

**Pull request merged**